blog/2024-10-29-vibration-analysis-python/index.mdx

```diff
@@ -0,0 +1,376 @@
+When conducted comprehensively and involves modern ML techniques, vibration analysis approaches the skill of an experienced mechanic’s trained ear, capable of sensing a mechanism's condition through subtle, often indescribable changes in its sound.
+However, let’s start with basic vibration analysis methods, specifically by calculating statistics or features that reflect certain well-interpreted properties of the process, each with clear physical meaning. This type of analysis addresses the most common questions about the process, while more subtle features - or those harder to formalize, or just specific for a particular case - may remain unqueried, despite their value for diagnostic and prognostic purposes.
+
+{/* truncate */}
+
+## Input variables
+
+Sensors of different types provide data on displacement, speed (the first derivative of displacement), and acceleration (the 2nd derivative). Acceleration is typically most sensitive to changes in the dynamic state, so, in most cases, it is the primary input for vibration analysis, which we'll focus on below.
+However, interpreting acceleration data in terms of actual machine part movement can be challenging. Reconstructing displacement by double integration of (potentially noisy) acceleration data is often impractical and unreliable. Therefore, in tasks where it is essential to monitor speed or the explicit trajectory of vibration, speed or displacement sensors (typically aligned along orthogonal axes) may be required.
+
+## Common features
+
+### Root Mean Square
+
+The simplest thing we might want to know about vibration signal is its "overall intensity", which can be formalized via the concept of energy: we may just consider its average energy within some time window. Energy, by its physical meaning, is a product of two primary (atomic, directly measurable) values: in mechanics, it's a product of force (or, better, acceleration or added velocity) and initial velocity, which leads to well-known energy formula with squared velocity, in electricity, energy is a product of voltage and current, that also can be converted to the formula with the squared current. To make our measure of intensity proportional to the amplitude of the signal, we can take the square root of average energy, i.e. get the constant value of amplitude that represents the same energy as the original signal within a time window. This measure is known as the **Root Mean Square (RMS)** of the signal x of length n:
+
+$$\text{RMS} = \sqrt{\frac{1}{n} \sum_{i=1}^{n} x_i^2}$$
```

Comment:
I think you can drop the electricity example "in electricity, energy is a product of voltage and current, that also can be converted to the formula with the squared current", instead, you could add the formula (Energy= 1/2 m v^2)

Reply:
- done